Stripe Integration

panikingginoo12 opened the pull request:
## Describe your changes
Added action set
- creating and retrieving customer information.
- attaching, retrieving and updating payment method.
- creating, updating, cancelling and retrieving of subscription.
- creating and retrieving of customer invoice.
- creating and retrieving of customer payment.

## Link to related issue

## Does this introduce new feature or change existing feature of Jaseci? If so, please add tests.

## Will this impact Jaseci users? If so, please write a paragraph describing the impact.

## Anything in this PR should the Jaseci developer/contributor community pay particular attention to? If so, please describe.

bcs-cristopher committed:
feat: initial commit for stripe action set

Added action set for creating and retrieving customer information. Added action set for attaching, retrieving and updating payment method. Added actino set for creating, cancelling and retrieving of subscription.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Jaseci-Labs/jaseci?pullRequest=385) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Jaseci-Labs/jaseci?pullRequest=385) before we can accept your contribution.<br/><hr/>**Cristopher Evangelista** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Jaseci-Labs/jaseci?pullRequest=385) it.</sub>

bcs-cristopher committed:
update: fix linting issue

bcs-cristopher committed:
Feat: added more stripe action set

Added action set
 - update_subscription
 - report_usage
 - create_invoice
 - get_invoice
 - get_customer_invoices
 - get_customer_payment
 - create_customer_payment

marsninja commented:
Hi @panikingginoo12 This implementation as an action set looks like a great idea of how to do it. Ultimately I think you can take the stripe_api out of the master and package it with the action. We can discuss once you're ready. Keep going though, this is a great direction.

bcs-cristopher committed:
Merge branch 'main' of https://github.com/Jaseci-Labs/jaseci into feature/stripe_action_set

panikingginoo12 commented:
Got it @marsninja I will do that. Thank you

bcs-cristopher committed:
update: moved stripe api to jaseci actions

bcs-cristopher committed:
Merge branch 'main' of https://github.com/Jaseci-Labs/jaseci into feature/stripe_action_set

bcs-cristopher committed:
fix: create_payment_intents payload

bcs-cristopher committed:
add: stripe.jac

bcs-cristopher committed:
feat: added stripe webhook initialization

bcs-cristopher committed:
Merge branch 'main' of https://github.com/Jaseci-Labs/jaseci into feature/stripe_action_set

bcs-cristopher committed:
Merge branch 'main' of https://github.com/Jaseci-Labs/jaseci into feature/stripe_action_set

bcs-cristopher committed:
update: move stripe to jaseci_serv

bcs-cristopher committed:
Merge branch 'main' of https://github.com/Jaseci-Labs/jaseci into feature/stripe_action_set

bcs-cristopher committed:
update: moved stripe stuff from jaseci_core to jaseci_serv

ypkang commented:
@panikingginoo12 @marsninja what are the next steps for this one?
